core/src/main/java/org/apache/iceberg/TableProperties.java

```diff
@@ -345,4 +345,11 @@ private TableProperties() {}
 
   public static final String UPSERT_ENABLED = "write.upsert.enabled";
   public static final boolean UPSERT_ENABLED_DEFAULT = false;
+
+  /**
+   * Table properties to be used when dropping a table. Contains a list of locations that are owned
+   * by the table and should also be dropped.
+   */
+  public static final String LOCATIONS_OWNED_BY_TABLE = "locations-owned-by-table";
```

Comment:
How would this property be populated? Does the user have to specify this explicitly per table? Or will a catalog auto-populate it?